### PR TITLE
ERR: Improve error message in validate_percentile

### LINKED ISSUE #51436: BUG: validate_percentile error message wrong with negative percentiles
```python
>>> pd.util._validators.validate_percentile([-1])
ValueError: percentiles should all be in the interval [0, 1]. Try [-0.01] instead.
```

-0.01 is still invalid.  The exception message should trim that suggestion.

### PR DESCRIPTION
**Pull request by ShashwatAgrawal20**
- [x] closes #51436 (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

This PR improves the validation of percentiles in pandas. Previously, when an out-of-range value was detected, the code would suggest a wrong value. This suggestion could be confusing to users and was not always necessary. This PR simplifies the suggestion message to directly state that the value should be between 0 and 1.

### CODE REVIEW

**Comment by ShashwatAgrawal20** [comment]
@jbrockmendel

File: pandas/util/_validators.py [hunk 1]
```diff
@@ -324,13 +324,13 @@ def validate_percentile(q: float | Iterable[float]) -> np.ndarray:
     q_arr = np.asarray(q)
     # Don't change this to an f-string. The string formatting
     # is too expensive for cases where we don't need it.
-    msg = "percentiles should all be in the interval [0, 1]. Try {} instead."
+    msg = "percentiles should all be in the interval [0, 1]"
     if q_arr.ndim == 0:
         if not 0 <= q_arr <= 1:
-            raise ValueError(msg.format(q_arr / 100.0))
```
Comment (phofl) [inline pandas/util/_validators.py:330]: Can we add a test for the new error message?

Comment (ShashwatAgrawal20) [reply pandas/util/_validators.py:330]: yeah !! 

> Can we add a test for the new error message?


What about something like this ?

```python
import re
import numpy as np
import pandas as pd
import pytest


@pytest.fixture
def valid_percentiles():
    return [
        [0.0, 0.25, 0.5, 0.75, 1.0],
        (0.0, 0.25, 0.5, 0.75, 1.0),
        np.array([0.0, 0.25, 0.5, 0.75, 1.0])
    ]


@pytest.fixture
def invalid_percentiles():
    return [
        [-0.01, 1.01],
        (-0.01, 1.01),
        np.array([-0.01, 1.01])
    ]


def test_validate_percentile_valid_input(valid_percentiles):
    for percentiles in valid_percentiles:
        result = pd.util._validators.validate_percentile(percentiles)
        assert np.array_equal(result, np.array(percentiles))


def test_validate_percentile_invalid_input(invalid_percentiles):
    expected_error_msg = "percentiles should all be in the interval [0, 1]"
    for p in invalid_percentiles:
        with pytest.raises(ValueError, match=re.escape(expected_error_msg)):
            pd.util._validators.validate_percentile(p)


def test_validate_percentile_negative_array_input():
    percentile_array = np.array([-0.5, 0.25, 1.5])
    expected_error_msg = "percentiles should all be in the interval [0, 1]"
    with pytest.raises(ValueError, match=re.escape(expected_error_msg)):
        pd.util._validators.validate_percentile(percentile_array)
```

Comment (phofl) [reply pandas/util/_validators.py:330]: Something like ``test_validate_percentile_negative_array_input`` is sufficient, you don't need the others I think.
can you trigger the error somehow from calling a public function? To check that the appropriate error is shown to the user.

Comment (ShashwatAgrawal20) [reply pandas/util/_validators.py:330]: > can you trigger the error somehow from calling a public function? To check that the appropriate error is shown to the user.

something like this ?
```python

import numpy as np
import pandas as pd

def my_function(percentiles):
    valid_percentiles = pd.util._validators.validate_percentile(percentiles)

percentiles = np.array([-0.5, 0.25, 1.5])
my_function(percentiles)

```

Comment (jbrockmendel) [reply pandas/util/_validators.py:330]: just call Series.quantile with the invalid values

Comment (ShashwatAgrawal20) [reply pandas/util/_validators.py:330]: something like this ?

```python
import pandas as pd
import numpy as np

percentiles = [-0.5, 0.25, 1.5]
s = pd.Series(np.arange(10))
try:
    s.quantile(percentiles)
except ValueError as e:
    print(str(e))
```

Comment (phofl) [reply pandas/util/_validators.py:330]: You can check the contributor guide on how we test exceptions (or check other tests)

Comment (ShashwatAgrawal20) [reply pandas/util/_validators.py:330]: I apologize for the delay in my response.

Hope something like this ?

```python
import pandas as pd
import numpy as np
import pytest

def test_series_quantile_invalid_percentile():
    s = pd.Series(np.random.randn(100))
    percentile_array = [-0.5, 0.25, 1.5]
    with pytest.raises(ValueError, match="percentiles should all be in the interval [0, 1]"):
        s.quantile(percentile_array)
```

Comment (phofl) [reply pandas/util/_validators.py:330]: Yep this looks good

Comment (ShashwatAgrawal20) [reply pandas/util/_validators.py:330]: Sorry, but I wonder how can I put the things together ?

Comment (phofl) [reply pandas/util/_validators.py:330]: Please read our contributing guide, there is an explanation how to add tests

**Review by phofl — CHANGES_REQUESTED** [review]
This isn't the test we agreed upon.

Please call the message simply msg and you don't need brackets after match=

**Comment by ShashwatAgrawal20** [comment]
I apologize for the mistake I made earlier.

File: pandas/tests/util/test_validator.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+import pandas as pd
+
+
+def test_series_quantile_invalid_percentile():
+    s = pd.Series(np.random.randn(100))
+    percentile_array = [-0.5, 0.25, 1.5]
+    msg = "percentiles should all be in the interval \\[0, 1\\]"
+    with pytest.raises(ValueError, match=msg):
+        s.quantile(percentile_array)
```
Comment (phofl) [inline pandas/tests/util/test_validator.py:12]: There is a file for quantile tests

Comment (ShashwatAgrawal20) [reply pandas/tests/util/test_validator.py:12]: The `pandas/tests/frame/methods/test_quantile.py` one ?

File: pandas/tests/series/methods/test_quantile.py [hunk 3]
```diff
@@ -38,10 +38,11 @@ def test_quantile(self, datetime_series):
         result = Series([np.timedelta64("NaT")]).sum()
         assert result == pd.Timedelta(0)
 
+        s = Series(np.random.randn(100))
+        percentile_array = [-0.5, 0.25, 1.5]
         msg = "percentiles should all be in the interval \\[0, 1\\]"
-        for invalid in [-1, 2, [0.5, -1], [0.5, 2]]:
```
Comment (phofl) [inline pandas/tests/series/methods/test_quantile.py:42]: Why are you deleting those?

Comment (ShashwatAgrawal20) [reply pandas/tests/series/methods/test_quantile.py:42]: Should I need append the `Series` test ?

**Comment by ShashwatAgrawal20** [comment]
@phofl, Is this ok?

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @ShashwatAgrawal20

**Pull request merged**